guide.md

```diff
@@ -154,54 +142,92 @@ enum FooBar {
 }
 ```
 
-Build an enum value as you would a namespaced struct, tuple struct, or constant.
-
 ### macro\_rules!
 
 Use `{}` for the full definition of the macro.
 
-DO NOT use spaces around the `:`.
+```rust
+macro_rules! foo {
+}
+```
 
-### String literals
+In patterns, put a space after, but not before a `:`:
```

Comment:
I disagree with this. `$x:ident` is far easier to read than `$x: ident`, because of things like:

```rust
macro_rules! foo {
    ($x: expr : $t: ty) => {}
}
```

as opposed to

```rust
macro_rules! foo {
    ($x:expr : $t:ty) => {}
}
```